spec.md

```diff
@@ -1081,6 +1092,34 @@ $instance = new class extends \Foo implements
 };
 ```
 
+## 9. Enumerations
+
+Enumerations (enums) MUST follow the same guidelines as classes, except where otherwise noted below.
+
+Methods in enums MUST follow the same guidelines as methods in classes.  Non-public methods MUST use `private`
+instead of `protected`, as enums do not support inheritance.
+
+When using a backed enum, there MUST NOT be a space between the enum name and colon, and there MUST be exactly one
+space between the colon and the backing type.  This is consistent with the style for return types.
+
+Enum case declarations MUST use CamelCase capitalization.  Enum case declarations MUST be on their own line.
+
+Constants in Enumerations MAY use either CamelCase or UPPER_CASE capitalization.  CamelCase is RECOMMENDED,
+so that it is consistent with case declarations.
```

Comment:
I believe you meant PascalCase? 
The Upper-CamelCase is not much used anymore, usually when camelCase is used it means the Lower-camelCase

Reply:
I see CamelCase used way more than PascalCase.  The vast majority of the programming world has never used Pascal. :smiley:

Reply:
```
camelCase
snake_case
PascalCase
```

I think @mabar is correct here. First-letter uppercase is PascalCase, not camelCase.

Reply:
Yes PSR-1 defines the following:
```
- Class names MUST be declared in `StudlyCaps`.
- Method names MUST be declared in `camelCase`.
```
and PSR-12 adds clarification:
```
The term 'StudlyCaps' in PSR-1 MUST be interpreted as PascalCase where the first letter of each word is capitalized including the very first letter.
```

so let's continue to use these terms where "camel case" means `fooBar` and "pascal case" means `FooBar`